<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>994</width>
    <height>750</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="OpenGLWidget" name="openGLWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>661</width>
      <height>541</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="pB_Generate">
    <property name="geometry">
     <rect>
      <x>690</x>
      <y>630</y>
      <width>141</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Generate</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pB_Clear">
    <property name="geometry">
     <rect>
      <x>840</x>
      <y>630</y>
      <width>141</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Clear</string>
    </property>
   </widget>
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>670</x>
      <y>20</y>
      <width>321</width>
      <height>501</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="Setting">
     <attribute name="title">
      <string>Setting</string>
     </attribute>
     <widget class="QGroupBox" name="groupBox">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>341</width>
        <height>501</height>
       </rect>
      </property>
      <property name="title">
       <string/>
      </property>
      <widget class="QLabel" name="l_TrunkLen">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>121</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Trunk length:</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="lE_TrunkLen">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>20</y>
         <width>131</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>5.5</string>
       </property>
      </widget>
      <widget class="QLabel" name="l_TrunkLen_ratio">
       <property name="geometry">
        <rect>
         <x>-10</x>
         <y>50</y>
         <width>171</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>   TrunkLength ratio :</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="lE_TrunkLen_ratio">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>50</y>
         <width>131</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>0.85</string>
       </property>
      </widget>
      <widget class="QLabel" name="l_TrunkR">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>90</y>
         <width>121</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Trunk radius:</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="LE_TrunkR">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>90</y>
         <width>131</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>0.3</string>
       </property>
      </widget>
      <widget class="QLabel" name="l_TrunkLenR_ratio">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>130</y>
         <width>151</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>TrunkRadius ratio:</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="lE_TrunkR_ratio">
       <property name="geometry">
        <rect>
         <x>162</x>
         <y>130</y>
         <width>131</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>0.75</string>
       </property>
      </widget>
      <widget class="QLabel" name="l_BerryR">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>170</y>
         <width>111</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Berry radius:</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="lE_BerryR">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>170</y>
         <width>131</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>2</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="lE_BezierN">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>210</y>
         <width>131</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>10</string>
       </property>
      </widget>
      <widget class="QLabel" name="l_BezierN">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>210</y>
         <width>91</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Bezier n :</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="lE_dx">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>250</y>
         <width>131</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>60</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>250</y>
         <width>91</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>dx:</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>290</y>
         <width>72</width>
         <height>15</height>
        </rect>
       </property>
       <property name="text">
        <string>dy:</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="lE_dy">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>290</y>
         <width>131</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>10</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_4">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>330</y>
         <width>72</width>
         <height>15</height>
        </rect>
       </property>
       <property name="text">
        <string>dz:</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="lE_dz">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>330</y>
         <width>131</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>60</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_6">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>360</y>
         <width>131</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Cylinder_step:</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="l_Cylinder_step">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>370</y>
         <width>131</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>30</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_7">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>400</y>
         <width>72</width>
         <height>15</height>
        </rect>
       </property>
       <property name="text">
        <string>lats:</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_8">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>430</y>
         <width>72</width>
         <height>15</height>
        </rect>
       </property>
       <property name="text">
        <string>longs:</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="l_lats">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>400</y>
         <width>131</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>30</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="l_longs">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>430</y>
         <width>131</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>30</string>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="Grammar">
     <attribute name="title">
      <string>Grammar</string>
     </attribute>
     <widget class="QTextBrowser" name="textBrowser_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>301</width>
        <height>351</height>
       </rect>
      </property>
      <property name="html">
       <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'SimSun'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; color:#008000;&quot;&gt;FB[/&amp;amp;UUS&amp;amp;FS]AWB[^^/USFS]AWW[*&amp;amp;FS]AW[//FS]AWW[&amp;amp;*FS]A$$C&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; color:#008000;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; color:#008000;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; color:#008000;&quot;&gt;FB[*^FS]AUB[/&amp;amp;UUS&amp;amp;FS]AW[^//FS]AW[&amp;amp;&amp;amp;US&amp;amp;FS]A$C&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-family:'Courier New'; color:#008000;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; color:#008000;&quot;&gt;FB[^US*FS]AWB[//US/FS]AW[/^^FS]A%%C&amp;quot;&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-family:'Courier New'; color:#008000;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; color:#008000;&quot;&gt;FB[/^US^FS]AUB[&amp;amp;*US*FS]AUW[*^^FS]AW[///FS]%C&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; color:#c0c0c0;&quot;&gt;     &lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; color:#008000;&quot;&gt;FB[//FS]AWB[&amp;amp;&amp;amp;FS]AUW[**FS]AWW[//&amp;amp;FS]A$C&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>390</y>
        <width>151</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Give new Grammar:</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lE_Grammar">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>390</y>
        <width>161</width>
        <height>21</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="pB_AddGra">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>420</y>
        <width>93</width>
        <height>28</height>
       </rect>
      </property>
      <property name="text">
       <string>Add</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="Information">
     <attribute name="title">
      <string>Default_Setting</string>
     </attribute>
     <widget class="QTextBrowser" name="textBrowser_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>281</width>
        <height>431</height>
       </rect>
      </property>
      <property name="html">
       <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'SimSun'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; color:#808000;&quot;&gt;double&lt;/span&gt;&lt;span style=&quot; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; color:#800000;&quot;&gt;length&lt;/span&gt;&lt;span style=&quot; color:#c0c0c0;&quot;&gt; &lt;/span&gt;=&lt;span style=&quot; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; color:#000080;&quot;&gt;5.5&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; color:#808000;&quot;&gt;double&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; color:#800000;&quot;&gt;length_shrink&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New';&quot;&gt;=&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; color:#000080;&quot;&gt;0.85&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New';&quot;&gt;;&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; color:#808000;&quot;&gt;double&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; color:#800000;&quot;&gt;radius&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New';&quot;&gt;=&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; color:#000080;&quot;&gt;0.3&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New';&quot;&gt;;&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; color:#808000;&quot;&gt;double&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; color:#800000;&quot;&gt;radius_shrink&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New';&quot;&gt;=&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; color:#000080;&quot;&gt;0.75&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New';&quot;&gt;;&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; color:#808000;&quot;&gt;double&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; color:#800000;&quot;&gt;leap_x&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New';&quot;&gt;=&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; color:#000080;&quot;&gt;60.00&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New';&quot;&gt;;&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; color:#808000;&quot;&gt;double&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; color:#800000;&quot;&gt;leap_y&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New';&quot;&gt;=&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; color:#000080;&quot;&gt;10.00&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New';&quot;&gt;;&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; color:#808000;&quot;&gt;double&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; color:#800000;&quot;&gt;leap_z&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New';&quot;&gt;=&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; color:#000080;&quot;&gt;60.00&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>Output</string>
     </attribute>
     <widget class="QPushButton" name="pB_Grammar">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>90</y>
        <width>171</width>
        <height>41</height>
       </rect>
      </property>
      <property name="text">
       <string>Output Grammar</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pB_Json">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>140</y>
        <width>171</width>
        <height>41</height>
       </rect>
      </property>
      <property name="text">
       <string>Output Model(Json)</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_5">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>50</y>
        <width>91</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>File name:</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lE_file">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>50</y>
        <width>151</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="pB_LoadJson">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>190</y>
        <width>171</width>
        <height>41</height>
       </rect>
      </property>
      <property name="text">
       <string>Load Model</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pB_Dp">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>300</y>
        <width>171</width>
        <height>41</height>
       </rect>
      </property>
      <property name="text">
       <string>Digital phantom</string>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QTextBrowser" name="textBrowser">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>560</y>
      <width>661</width>
      <height>111</height>
     </rect>
    </property>
   </widget>
   <widget class="QCheckBox" name="checkBox">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>520</y>
      <width>191</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Disable Berry</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pB_Generate_2">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>590</y>
      <width>151</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>wine grape</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pB_Generate_3">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>560</y>
      <width>151</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>table grape</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>560</y>
      <width>131</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>berry color</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>994</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>OpenGLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header location="global">openglwidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
